src/Analyzers/Core/Analyzers/SimplifyInterpolation/Helpers.cs

```diff
@@ -132,6 +136,59 @@ private static IOperation Unwrap(IOperation expression)
             formatString = null;
         }
 
+        private static bool IsObjectToStringOverride(IMethodSymbol method)
+        {
+            while (method.OverriddenMethod is not null)
+                method = method.OverriddenMethod;
+
+            return method.ContainingType.SpecialType == SpecialType.System_Object
+                && method.Name == nameof(ToString);
+        }
+
+        private static bool IsInvariantCultureReference(IOperation operation)
+        {
+            if (Unwrap(operation) is not IPropertyReferenceOperation { Member: { } member })
+                return false;
+
+            return member.Name switch
+            {
+                nameof(CultureInfo.InvariantCulture) => SymbolEqualityComparer.Default.Equals(
+                    member.ContainingType,
+                    operation.SemanticModel!.Compilation.GetTypeByMetadataName(typeof(System.Globalization.CultureInfo).FullName!)),
```

Comment:
pull out the compilation above and bail if it is null.

Reply:
If the compilation is null, shouldn't I throw ArgumentException because the helper is not capable of doing its job if the caller passes in ops that have no semantic model?

Maybe I should instead pass down the compilation as an additional method arg.

Reply:
Not happy with how the code bulks up if I pass it down, and `invocation.SemanticModel!.Compilation` is already used in UnwrapFormatString.

Reply:
Pushed a commit using Contract.ThrowIfNull as discussed in chat, but I'd still prefer docs + ArgumentException because the caller is at fault if they try to get an answer from the helper without providing a semantic model.

Reply:
That's what Contract is for.  An invariant violation. Argument exceptions are for validating or pubic surface. The represent our contract to the outside world. Contract is for stating it assumptions for how a piece of code is intended to be used.